src/coreclr/src/jit/optcse.cpp

```diff
@@ -2696,7 +2696,9 @@ class CSE_Heuristic
             GenTree*      cse = nullptr;
             bool          isDef;
             FieldSeqNode* fldSeq               = nullptr;
-            bool          hasZeroMapAnnotation = m_pCompiler->GetZeroOffsetFieldMap()->Lookup(exp, &fldSeq);
+            bool          commaOnly            = true;
+            GenTree*      effectiveExp         = exp->gtEffectiveVal(commaOnly);
+            const bool    hasZeroMapAnnotation = m_pCompiler->GetZeroOffsetFieldMap()->Lookup(effectiveExp, &fldSeq);
```

Comment:
It looks like you deleted the place below that uses this bool, is that intentional?